SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -32,20 +34,32 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
     }
 
     override fun onButtonTapped() {
+        if (chosenDepartureStation == chosenArrivalStation) return
         val dateTimeFormat = DateFormat("yyyy-MM-ddTHH:mm:ss.000")
         val timeNow: String = DateTimeTz.nowLocal().format(dateTimeFormat)
 
         val apiCall = "https://mobile-api-dev.lner.co.uk/v1/fares?originStation=$chosenDepartureStation&destinationStation=$chosenArrivalStation&noChanges=false&numberOfAdults=1&numberOfChildren=0&journeyType=single&outboundDateTime=$timeNow&outboundIsArriveBy=false"
+
         val client = HttpClient() {
             install(JsonFeature) {
                 serializer = KotlinxSerializer(Json.nonstrict)
             }
         }
 
         launch {
-            val jsonString = client.get<DepartureDetails>(apiCall)
+            lateinit var jsonString: DepartureDetails
```

Comment:
`jsonString` is a slightly misleading name, since it's not actually a string because you've mapped it to a DepartureDetails objects, do you agree?